src/node/txdownloadman_impl.cpp

```diff
@@ -301,6 +301,10 @@ void TxDownloadManagerImpl::MempoolAcceptedTx(const CTransactionRef& tx)
 
 node::RejectedTxTodo TxDownloadManagerImpl::MempoolRejectedTx(const CTransactionRef& ptx, const TxValidationState& state, NodeId nodeid, bool first_time_failure)
 {
+    if (!Assume(state.IsInvalid())) {
```

Comment:
this is definitely problematic, but I can't seem to figure out a way to hit it. In prod this would "just" cause it to not call `m_txrequest.ForgetTxHash(ptx->GetWitnessHash());` for it, but would have to be in a 1p1c package setting.

To retain current behavior, I think it can just be expanded to 
```suggestion
    // DIFFERENT_WITNESS has "valid" status, but we still want to stop requesting it
    if (!Assume(state.IsInvalid() ||
        tx_result.m_result_type == MempoolAcceptResult::ResultType::DIFFERENT_WITNESS)) {
```

Reply:
Good point - I feel like we actually shouldn't be adding the `Assume(IsInvalid())` here then. We don't have the `MempoolAcceptResult` here (and we shouldn't do that imo as txdownloadman.h would have a dependency on validation.h)